lib/autobahn/consumer.rb

```diff
@@ -121,10 +120,17 @@ def setup!
     end
 
     def deliver(handler, timeout)
-      begin
+      while @deliver.get
         headers, message = self.next(timeout)
         handler.call(headers, message) if headers
-      end while @deliver.get
+      end
+      while (headers, message = self.next(0))
+        if headers
+          handler.call(headers, message)
+        else
+          break
+        end
+      end
```

Comment:
Is there any real reason to use the public next method here? Couldn't we use the demultiplexer directly?
```ruby
while @deliver.get
  if (entry = @demultiplexer.take(timeout))
    handler.call(*entry)
  end
end
while (entry = @demultiplexer.take(0))
  handler.call(*entry)
end
```